osbenchmark/synthetic_data_generator/mapping_synthetic_data_generator.py

```diff
@@ -0,0 +1,438 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+import json
+import random
+import datetime
+import uuid
+import os
+import time
+import logging
+import hashlib
+from typing import Dict, Any, Callable
+
+import yaml
+from dask.distributed import Client, get_client
+from mimesis import Generic
+from mimesis.locales import Locale
+from mimesis.random import Random
+from tqdm import tqdm
+
+from osbenchmark.utils import console
+import osbenchmark.exceptions
+from osbenchmark.synthetic_data_generator.types import SyntheticDataGeneratorConfig, GB_TO_BYTES
+from osbenchmark.synthetic_data_generator.helpers import get_generation_settings, write_chunk, setup_custom_tqdm_formatting
+
+class MappingSyntheticDataGenerator:
+    def __init__(self, mapping_config=None):
+        self.logger = logging.getLogger(__name__)
+        # TODO: Set self.mapping_config to automatically point to MappingSyntheticDataGenerator
+        self.mapping_config = mapping_config or {}
+        # self.locale = self.mapping_config.get('mimesis_locale', 'DEFAULT')
+        seed = 1
+        self.generic = Generic(locale=Locale.EN)
+        self.random = Random()
+
+        self.generic.reseed(seed)
+        self.random.seed(seed)
+
+        # seed these
+        self.type_generators = {
+            "text": self.generate_text,
+            "keyword": self.generate_keyword,
+            "long": self.generate_long,
+            "integer": self.generate_integer,
+            "short": self.generate_short,
+            "byte": self.generate_byte,
+            "double": self.generate_double,
+            "float": self.generate_float,
+            "boolean": self.generate_boolean,
+            "date": self.generate_date,
+            "ip": self.generate_ip,
+            "object": self.generate_object,
+            "nested": self.generate_nested,
+            "geo_point": self.generate_geo_point,
+        }
+
+    def generate_text(self, field_def: Dict[str, Any],  **params) -> str:
+        choices = params.get('must_include', None)
+        analyzer = field_def.get("analyzer", "standard")
+
+        #TODO: Need to support other analyzers
+        text = ""
+        if choices:
+            term = random.choice(choices)
+            text += f"{term} "
+        if analyzer == "keyword":
+            text += f"keyword_{uuid.uuid4().hex[:8]}"
+            return text
+
+        text += f"Sample text for {random.randint(1, 100)}"
+        return text
+
+
+    def generate_keyword(self, field_def: Dict[str, Any], **params) -> str:
```

Comment:
all these can be `@property`, right?

Reply:
`@property` is more appropriate for when we need to representing class or instance attributes and modifying (get/set/delete) them. Should keep these as methods since they're not using any class / instance attributes and each time should generate different data.